Dockerfile

```diff
@@ -13,6 +13,9 @@ COPY cmd cmd
 COPY api api
 COPY internal internal
 COPY provisioner provisioner
+# copy git-related information for binary versioning
+COPY .git .git
```

Comment:
Hmm - do we need to copy the entire .git to be able to access version information? Can we supply this Dockerfile with a build-arg variable instead, and that variable can derive the requisite git information before the build stage?

Reply:
Thinking about this some more: I'm mainly interested in copying over the entire .git directory due to increasing build times if that directory is bloated (and/or contains sensitive information). If we don't like the approach of using a build argument to drive version information to the go binaries, can we exclude everything in the .git directory that we _don't_ need to parse the version information in the .dockerignore?

Reply:
Yeah, it makes sense. I tried copying over just the minimum like in OLM's Dockerfile:
```
COPY .git/HEAD .git/HEAD
COPY .git/refs/heads/. .git/refs/heads
RUN mkdir -p .git/objects
```

but then I see this error in the container build logs: `fatal: HEAD is neither a commit nor blob` -- which suggests that the git tree is messed up somehow and I'm not sure how to fix it.

Reply:
We could change the version field to just capture the git commit, like in OLM and then the above would work. 
```
VERSION ?= $(shell git rev-parse HEAD)
```
we lose some metadata about the tag and number of commits since the tag though.

Reply:
Side note - this made me realize that Combo's version in its local container is not working so I'm making an issue against that repo.